app/src/main/java/com/team41/boromi/controllers/AuthenticationController.java

```diff
@@ -0,0 +1,104 @@
+package com.team41.boromi.controllers;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.team41.boromi.callbacks.AuthCallback;
+import com.team41.boromi.dagger.ActivityScope;
+import com.team41.boromi.dagger.BoromiModule;
+import com.team41.boromi.dbs.UserDB;
+import com.team41.boromi.models.User;
+import java.util.concurrent.Executor;
+import javax.inject.Inject;
+
+/**
+ * A class that handles authentication and sign in methods
+ */
+@ActivityScope
+public class AuthenticationController {
+
+  private final static String TAG = "AuthController";
+  final Executor executor;
+  UserDB userDB;
+  FirebaseAuth auth;
+
+  @Inject
+  public AuthenticationController(UserDB userDB, FirebaseAuth auth, Executor executor) {
+    this.userDB = userDB;
+    this.auth = auth;
+    this.executor = executor;
+  }
+
+  // TODO
+  // IMPLEMENTATION OF LOGIN WITH USERNAME
```

Comment:
I clarified with prof that login with username isn't required. 
Login with email is sufficient since login with username is not straightforward in firebase

Reply:
Sounds good !